raft/src/main/java/org/apache/kafka/raft/internals/KafkaRaftMetrics.java

```diff
@@ -71,10 +71,14 @@ public KafkaRaftMetrics(Metrics metrics, String metricGrpPrefix, QuorumState sta
         Gauge<String> stateProvider = (mConfig, currentTimeMs) -> {
             if (state.isLeader()) {
                 return "leader";
+            } else if (state.isProspectiveNotVoted()) {
+                return "prospective";
+            } else if (state.isProspectiveAndVoted()) {
+                return "prospective-voted";
             } else if (state.isCandidate()) {
                 return "candidate";
             } else if (state.isUnattachedAndVoted()) {
-                return "voted";
+                return "unattached-voted";
```

Comment:
This is a publicly visible change. Let's update the KIP if it doesn't include this change.